encoding.bs

```diff
@@ -201,10 +201,17 @@ an I/O queue "<code>&amp;#128169; hello world</code>". The next item to be read
 <p>To <dfn for="from I/O queue">convert</dfn> an <a for=/>I/O queue</a> <var>ioQueue</var> into a
 <a for=/>list</a>, <a>string</a> or <a>byte sequence</a>, return the result of
 <a for="I/O queue">reading</a> an indefinite number of <a for=list>items</a> from
-<var>ioQueue</var>. To <dfn for="to I/O queue">convert</dfn> a <a for=/>list</a>, <a>string</a> or
-<a>byte sequence</a> <var>input</var> into an <a for=/>I/O queue</a>, return an
-<a for=/>I/O queue</a> containing the <a for=list>items</a> in <var>input</var>, in order, followed
-by <a>end-of-queue</a>.
+<var>ioQueue</var>.
+
+<p>To <dfn for="to I/O queue">convert</dfn> a <a for=/>list</a>, <a>string</a> or
+<a>byte sequence</a> <var>input</var> into an <a for=/>I/O queue</a>, run these steps:
+
+<ol>
+ <li><p>Assert: <var>input</var> does not <a for=list>contain</a> <a>end-of-queue</a>.
```

Comment:
This needs a conditional for input being a list. Or we do it similar to prepend and say "a list not containing end-of-queue" in the algorithm "header".

Reply:
I tried having the assertion in the algorithm header, and it did not flow well ("to convert one of a list not containing end-of-string, a string or a byte sequence `input`"). But sure, I'll add the conditional.